src/fluree/db/connection.cljc

```diff
@@ -20,6 +20,22 @@
 ;; socket connections are keyed by connection-id and contain :socket - ws, :id - socket-id :health - status of health checks.
 (def server-connections-atom (atom {}))
 
+;; regex to parse URI
+;; Returns an array of strings in order of:
+;; 0 - href  e.g., "https://my-fluree-server.ee:8090/fdb/test/chat/query"
+;; 1 - protocol  e.g., "https:"
+;; 2 - hostname  e.g., "my-fluree-server.ee"
+;; 3 - port  e.g., "8090"
+;; 4 - pathname  e.g. "fdb/test/chat/query"
+;; 5 - search  e.g., "?example=notsupported" if "https://my-fluree-server.ee:8090/fdb/test/chat/query?example=notsupported"
+;; 6 - hash  e.g., "#authority" if https://docs.flur.ee/guides/identity/auth-records#authority
+(def uri-regex #?(:clj #"^(?:((?:https?|s?ftp):)\/\/)([^:\/\s]+)(?::(\d*))?(?:\/([^\s?#]+)?([?][^?#]*)?(#.*)?)?"
+                  :cljs #"^(?:((?:https?|s?ftp):)//)([^:/\s]+)(?::(\d*))?(?:/([^\s?#]+)?([?][^?#]*)?(#.*)?)?"))
+(def server-regex #?(:clj #"^(?:((?:https?):)\/\/)([^:\/\s#]+)(?::(\d*))?"
+                     :cljs #"^(?:((?:https?):)//)([^:/\s#]+)(?::(\d*))?"))
```

Comment:
Does the re-pattern take care of the differences between Clojure (query peer) and ClojureScript (JavaScript libraries)?  Because the / needs to be escaped in Clojure string but the clojurescript build automatically adds the escape character while parsing the string.

Reply:
I don't know about the repl; but, it actually fails in the nodejs/cljs test builds:

May 11, 2021 3:38:48 PM com.google.javascript.jscomp.LoggerErrorManager println
SEVERE: /Users/lois/fluree/db/out/nodejs/build/fluree/db/connection.js:18:59: ERROR - [JSC_PARSE_ERROR] Parse error. Invalid escape sequence
  18| fluree.db.connection.server_regex = /^(?:((?:https?):)\\/\\/)([^:\\/\s#]+)(?::(\d*))?/;
                                                                 ^

May 11, 2021 3:38:48 PM com.google.javascript.jscomp.LoggerErrorManager printSummary
WARNING: 1 error(s), 0 warning(s)
ERROR: JSC_PARSE_ERROR. Parse error. Invalid escape sequence at /Users/lois/fluree/db/out/nodejs/build/fluree/db/connection.js line 18 : 59
Unexpected error compiling at (REPL:1).
Closure compilation failed

Full report at:
/var/folders/09/npgfz9kd62n4kflm_xxrt_7c0000gn/T/clojure-10593699393783293925.edn
make: *** [out/flureenjs.js] Error 1